ruby/publish/action.yml

```
@@ -0,0 +1,156 @@
+    description: The template for the release message. Use "{0}" in the text to refer to the current version.
+    required: true
+  silk_asset_group:
+    description: The Silk asset group for the project
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: "Create temporary app token"
+      uses: actions/create-github-app-token@v1
+      id: app-token
+      with:
+        app-id: ${{ inputs.app_id }}
+        private-key: ${{ inputs.app_private_key }}
+
+    - name: "Store GitHub token in environment"
+      shell: bash
+      run: echo "GH_TOKEN=${{ steps.app-token.outputs.token }}" >> "$GITHUB_ENV"
+
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        token: ${{ env.GH_TOKEN }}
+
+    - name: Setup Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: '3.2'
+        bundler-cache: true
+
+    - name: Get the release version
+      shell: bash
+      run: echo "RELEASE_VERSION=$(bundle exec rake version)" >> "$GITHUB_ENV"
+
+    - name: Setup GitHub tooling for DBX Drivers
+      uses: mongodb-labs/drivers-github-tools/setup@v2
+      with:
+        aws_role_arn: ${{ inputs.aws_role_arn }}
+        aws_region_name: ${{ inputs.aws_region_name }}
+        aws_secret_id: ${{ inputs.aws_secret_id }}
+
+    - name: Set output gem file name
+      shell: bash
+      run: |
+        echo "GEM_FILE_NAME=${{ inputs.gem_name }}-${{ env.RELEASE_VERSION }}.gem" >> "$GITHUB_ENV"
+
+    - name: Build the gem
+      shell: bash
+      run: |
+        gem build --output=${{ env.GEM_FILE_NAME }} ${{ inputs.gem_name }}.gemspec
+
+    - name: Sign the gem
+      uses: mongodb-labs/drivers-github-tools/gpg-sign@v2
+      with:
+        filenames: '${{ env.GEM_FILE_NAME }}'
+
+    - name: Generate authorized publication Documentation
+      uses: mongodb-labs/drivers-github-tools/authorized-pub@v2
+      with:
+        product_name: ${{ inputs.product_name }}
+        release_version: ${{ env.RELEASE_VERSION }}
+        filenames: ${{ env.GEM_FILE_NAME }}
+        token: ${{ env.GH_TOKEN }}
+
+    - name: Download SBOM file from Silk
+      uses: mongodb-labs/drivers-github-tools/sbom@v2
+      with:
+        silk_asset_group: ${{ inputs.silk_asset_group }}
+
+    - name: Generate Sarif Report
+      uses: mongodb-labs/drivers-github-tools/code-scanning-export@v2
+      with:
+        output-file: ${{ env.S3_ASSETS }}/code-scanning-alerts.json
+
+    - name: Generate Compliance Report
+      uses: mongodb-labs/drivers-github-tools/compliance-report@v2
+      with:
+        token: ${{ env.GH_TOKEN }}
+
+    - name: Create and sign the tag
+      uses: mongodb-labs/drivers-github-tools/git-sign@v2
+      with:
+        command: "git tag -u ${{ env.GPG_KEY_ID }} -m 'Release tag for v${{ env.RELEASE_VERSION }}' v${{ env.RELEASE_VERSION }}"
+
+    - name: Push the tag to the repository
+      shell: bash
+      run: |
+        git push origin v${{ env.RELEASE_VERSION }}
```

The `tag-version` action handles these two steps:

```suggestion
    - name: "Create release tag"
      uses: mongodb-labs/drivers-github-tools/tag-version@v2
      with:
        tag_template: "v${VERSION}"
        tag_message_template: "Release tag for v${VERSION}"
```

Note that you may not want to push the tag if this is a dry run. In that case, the `push_tag` option allows you to disable the corresponding `git push`. I'm now also realising that my suggestion to make this option the opposite of `dry_run` is making this a little more complicated, especially since GHA expressions don't support ternary operators. The input would look something like this:
`push_tag: ${{ inputs.dry_run == 'false' && 'true' || 'false' }}`